### PR TITLE
feat: Merge upstream changes from upstream release/v0.19.x branch [DEV-2776]

### PR DESCRIPTION
**Pull request by ankurdotb**

### CODE REVIEW

File: benchmarks/cosmos-exim/main.go [hunk 1]
```diff
@@ -128,7 +128,10 @@ func runExport(dbPath string) (int64, map[string][]*iavl.ExportNode, error) {
 			return 0, nil, err
 		}
 		start := time.Now().UTC()
-		exporter := itree.Export()
+		exporter, err := itree.Export()
+		if err != nil {
+			return 0, nil, err
+		}
```
Comment (ankurdotb) [inline benchmarks/cosmos-exim/main.go:134]: IIRC, this is bit that was causing panic/failure last time. However, see other files/changes holistically.

File: diff.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
+package iavl
```
Comment (ankurdotb) [inline diff.go:1]: Commenting on the whole file: this is entirely new, so assess

File: benchmarks/cosmos-exim/main.go (hunk 1 above)
Comment (Eengineer1) [reply benchmarks/cosmos-exim/main.go:134]: No, it was in `export.go`, as per below.

**Comment by ankurdotb** [comment]
Task linked: [DEV-2776 Assess whether we can use vanilla comos/iavl or need to patch our fork](https://app.clickup.com/t/6600954/DEV-2776)

**Pull request closed without merging**